dbms/src/Storages/KVStore/MultiRaft/Disagg/FastAddPeer.cpp

```diff
@@ -185,26 +191,44 @@ std::variant<CheckpointRegionInfoAndData, FastAddPeerRes> FastAddPeerImplSelect(
     Stopwatch watch;
     std::unordered_map<StoreID, UInt64> checked_seq_map;
     auto fap_ctx = tmt.getContext().getSharedContextDisagg()->fap_context;
+    auto cancel_handle = fap_ctx->tasks_trace->getCancelHandleFromExecutor(region_id);
+
+    if (cancel_handle->canceled())
+    {
+        // It is already canceled in queue.
+        return genFastAddPeerRes(FastAddPeerStatus::Canceled, "", "");
+    }
+
+    // Get candidate stores.
     const auto & settings = tmt.getContext().getSettingsRef();
-    auto current_store_id = tmt.getKVStore()->getStoreMeta().id();
-    auto candidate_store_ids = getCandidateStoreIDsForRegion(tmt, region_id, current_store_id);
+    auto current_store_id = tmt.getKVStore()->clonedStoreMeta().id();
+    std::vector<StoreID> candidate_store_ids;
+    fiu_do_on(FailPoints::force_set_fap_candidate_store_id, { candidate_store_ids = {1234}; });
+    if (candidate_store_ids.empty())
+        candidate_store_ids = getCandidateStoreIDsForRegion(tmt, region_id, current_store_id);
```

Comment:
```suggestion
    std::vector<StoreID> candidate_store_ids = getCandidateStoreIDsForRegion(tmt, region_id, current_store_id);
    fiu_do_on(FailPoints::force_set_fap_candidate_store_id, { candidate_store_ids = {1234}; });
```

Reply:
`getCandidateStoreIDsForRegion` could throw, this failpoint is to avoid this throw while provide a suitable value for `candidate_store_ids `